### PR TITLE
Added APIs for visualiser operations

### PR DESCRIPTION
**Pull request by vikiCoder**
Added `pm.visualiser` APIs with following methods:

- `pm.visualiser.set` - Allow setting visualiser template and its options
- `pm.visualiser.clear` - Removes previously set visualiser template and its options

### CODE REVIEW

File: lib/sandbox/visualiser.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    /**
+     * Set visualiser template and its options
+     *
+     * @param {String} template - visualisation layout in form of template
+     * @param {Object} options - options to use while processing the template
+     * @param {Object} options.data - data object to be used in template
+     * @param {Object} options.handlebarsOptions - hanadlebars options to use while compiling template
+     */
+    set: function (template, options) {
+        this.return.visualiser = {
```
Comment (codenirvana) [inline lib/sandbox/visualiser.js]: `this` is confusing.

File: lib/sandbox/pmapi.js [hunk 2]
```diff
@@ -110,9 +111,19 @@ Postman = function Postman (bridge, execution, onRequest, cookieStore) {
...
+        /**
+         * @type {Object}
+         */
+        visualiser: Visualiser
     });
 
+    // bind pm.visualiser functions to execution instance
+    _.forEach(this.visualiser, function (value, key) {
+        typeof value === 'function' && (this.visualiser[key] = value.bind(execution));
+    }.bind(this));
```
Comment (codenirvana) [inline lib/sandbox/pmapi.js]: don't `bind`

File: lib/sandbox/pmapi.js [hunk 3]
```diff
@@ -110,7 +110,34 @@ Postman = function Postman (bridge, execution, onRequest, cookieStore) {
...
+         */
+        visualiser: {
+            /**
+             * Set visualiser template and its options
+             *
+             * @param {String} template - visualisation layout in form of template
+             * @param {Object} [data] - data object to be used in template
+             * @param {Object} [options] - options to use while processing the template
+             */
+            set: function (template, data, options) {
```
Comment (saswatds) [inline lib/sandbox/pmapi.js]: Need to enforce data and options are objects

Comment (vikiCoder) [reply lib/sandbox/pmapi.js]: How about we do this check in runtime while processing template?
If not then what should be done? ignore it or throw error?

File: test/unit/sandbox-libraries/pm.test.js [hunk 4]
```diff
@@ -813,6 +813,48 @@ describe('sandbox library - pm api', function () {
...
+            context.execute(`
+                var assert = require('assert');
+
+                assert.ok(pm.visualiser);
+                assert.strictEqual(typeof pm.visualiser.set, 'function');
+                assert.strictEqual(typeof pm.visualiser.clear, 'function');
+            `, {context: sampleContextData}, done);
+        });
+
+        it('pm.visualiser.set', function (done) {
```
[the remark this answers is not in the document]
Comment (vikiCoder) [reply test/unit/sandbox-libraries/pm.test.js]: I think these tests should be added in runtime while processing template.
Sandbox is simply passing whatever data provided by the user to runtime.

File: lib/sandbox/pmapi.js (hunk 3 above)
Comment (saswatds) [reply lib/sandbox/pmapi.js]: Throw error

File: lib/sandbox/pmapi.js [hunk 5]
```diff
@@ -110,7 +110,46 @@ Postman = function Postman (bridge, execution, onRequest, cookieStore) {
...
+
+                if (data && typeof data !== 'object') {
+                    throw new Error(`Invalid data. Data must be an object, found ${typeof data}`);
+                }
+
+                if (options && typeof options !== 'object') {
+                    throw new Error(`Invalid options. Options must be an object, found ${typeof options}`);
+                }
+
+                execution.return.visualizer = {
```
Comment (shamasis) [inline lib/sandbox/pmapi.js]: typedef mixin jsdoc here

**Pull request merged**